subsys/bluetooth/host/keys.h

```diff
@@ -46,7 +46,17 @@ struct bt_csrk {
 struct bt_keys {
 	u8_t                    id;
 	bt_addr_le_t		addr;
-	u8_t                    storage_start[0];
+#if defined (CONFIG_BT_KEY_DISPLACEMENT)
+#if !defined (CONFIG_BT_KEY_DISPLACEMENT_CNTR_IN_SETTINGS)
```

Comment:
You can combine the above two lines into a single `#if ...` statement by using `&&` between the two `defined(..)` statements. That way you only need a single `#endif` to terminate it.